mqtt-protocol/src/multithreadMain/kotlin/it/unibo/pulvreakt/mqtt/MqttProtocol.kt

```diff
@@ -1,62 +1,60 @@
+@file:Suppress("detekt:ImportOrdering")
+
 package it.unibo.pulvreakt.mqtt
 
+import MQTTClient
 import arrow.core.Either
-import arrow.core.raise.either
 import arrow.core.raise.ensureNotNull
+import arrow.core.raise.either
 import arrow.core.right
 import io.github.oshai.kotlinlogging.KotlinLogging
 import it.unibo.pulvreakt.api.communication.protocol.Entity
 import it.unibo.pulvreakt.api.communication.protocol.Protocol
-import it.unibo.pulvreakt.api.context.Context
 import it.unibo.pulvreakt.errors.protocol.ProtocolError
 import kotlinx.coroutines.CoroutineDispatcher
 import kotlinx.coroutines.CoroutineScope
 import kotlinx.coroutines.Job
-import kotlinx.coroutines.async
 import kotlinx.coroutines.cancelChildren
 import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.MutableSharedFlow
 import kotlinx.coroutines.flow.asSharedFlow
 import kotlinx.coroutines.launch
-import org.eclipse.paho.mqttv5.client.IMqttToken
-import org.eclipse.paho.mqttv5.client.MqttAsyncClient
-import org.eclipse.paho.mqttv5.client.MqttCallback
-import org.eclipse.paho.mqttv5.client.MqttConnectionOptions
-import org.eclipse.paho.mqttv5.client.MqttDisconnectResponse
-import org.eclipse.paho.mqttv5.client.persist.MemoryPersistence
-import org.eclipse.paho.mqttv5.common.MqttException
-import org.eclipse.paho.mqttv5.common.MqttMessage
-import org.eclipse.paho.mqttv5.common.packet.MqttProperties
+import mqtt.MQTTVersion
+import mqtt.Subscription
+import mqtt.packets.Qos
+import mqtt.packets.mqttv5.MQTT5Properties
+import mqtt.packets.mqttv5.ReasonCode
+import mqtt.packets.mqttv5.SubscriptionOptions
 import org.kodein.di.DI
-import org.kodein.di.instance
+import socket.IOException
 
 /**
- * MQTT Protocol implementation on JVM side.
+ * MQTT Protocol implementation on Native and JVM side.
  */
+@OptIn(ExperimentalUnsignedTypes::class)
+@Suppress("ExperimentalUnsignedTypes")
```

Comment:
Can we avoid the Unsigned Types?

Reply:
no, they come from the usage of UbyteArray in the KMQTT library for almost all the functions